### ISSUE TITLE
Yotta dependency project_generator setup.py is broken

### ISSUE THREAD

**Issue by carlosperate** [carlosperate/microbit-dev-env#5]
Thanks for the report @AlasdairAtKitronik.

When trying to install Yotta:

```
microbit-vg: #
microbit-vg: #
microbit-vg: # Installing yotta...
microbit-vg: #
microbit-vg: #
microbit-vg: + sudo -H pip install yotta
microbit-vg: Collecting yotta
microbit-vg:   Downloading https://files.pythonhosted.org/packages/d8/71/28998f9a11f2a530023ce927a5e0c274adada3f42289ebb5e18a50738369/yotta-0.18.5.tar.gz (130kB)
microbit-vg: Collecting semantic_version<3,>=2.3.1 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/72/83/f76958017f3094b072d8e3a72d25c3ed65f754cc607fdb6a7b33d84ab1d5/semantic_version-2.6.0.tar.gz
microbit-vg: Collecting requests[security]<3,>=2.4.3 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/49/df/50aa1999ab9bde74656c2919d9c0c085fd2b3775fd3eca826012bef76d8c/requests-2.18.4-py2.py3-none-any.whl (88kB)
microbit-vg: Collecting certifi>=2016.2.28 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/7c/e6/92ad559b7192d846975fc916b65f667c7b8c3a32bea7372340bfe9a15fa5/certifi-2018.4.16-py2.py3-none-any.whl (150kB)
microbit-vg: Collecting PyGithub<2,>=1.25 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/fd/28/c42bdd56111dae8c622c01af69e6d7f3d3ff0a96ad846f221749701f86b1/PyGithub-1.39.tar.gz (2.6MB)
microbit-vg: Collecting colorama<0.4,>=0.3 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/db/c8/7dcf9dbcb22429512708fe3a547f8b6101c0d02137acbd892505aee57adf/colorama-0.3.9-py2.py3-none-any.whl
microbit-vg: Collecting hgapi<2,>=1.7 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/36/db/6ad72214343e361c3fae732230e64bb7a5e4580002027782a4406748aee6/hgapi-1.7.4.tar.gz
microbit-vg: Collecting Jinja2<3,>=2.7.0 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/7f/ff/ae64bacdfc95f27a016a7bed8e8686763ba4d277a78ca76f32659220a731/Jinja2-2.10-py2.py3-none-any.whl (126kB)
microbit-vg: Collecting cryptography<2.2,>=2.1.4 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/69/7b/05067e7c4d1a287c0be040e46ef70b8dfd7a809af8afcfbcc1b60e801ebc/cryptography-2.1.4-cp27-cp27mu-manylinux1_x86_64.whl (2.2MB)
microbit-vg: Collecting PyJWT<2.0,>=1.0 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/31/8f/19c302aa9a391dd1fbd249362b749021b88d40fb59af0363939a2250afed/PyJWT-1.6.1-py2.py3-none-any.whl
microbit-vg: Collecting pathlib<1.1,>=1.0.1 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/ac/aa/9b065a76b9af472437a0059f77e8f962fe350438b927cb80184c32f075eb/pathlib-1.0.1.tar.gz (49kB)
microbit-vg: Collecting jsonschema<3.0,>=2.4.0 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/77/de/47e35a97b2b05c2fadbec67d44cfcdcd09b8086951b331d82de90d2912da/jsonschema-2.6.0-py2.py3-none-any.whl
microbit-vg: Collecting argcomplete<2.0,>=0.8.0 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/31/88/ba8d8684a8a27749250c66ff7c2b408fdbc29b50da61200338ff9b2607bf/argcomplete-1.9.4-py2.py3-none-any.whl
microbit-vg: Collecting mbed_test_wrapper<2.0.0,>=0.0.3 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/de/68/8c3af18cd03ce4c918e67cc5afa38c5b535b7238c594b8438f58e1e1bcf6/mbed_test_wrapper-1.0.0.tar.gz
microbit-vg: Collecting valinor<1.0,>=0.0.0 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/e6/5d/9f94c98fed3d3e003425aa50f29939c0ce935f99f81b7c21c653c52aef80/valinor-0.0.15.tar.gz
microbit-vg: Collecting intelhex<3.0,>=2.0 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/bf/77/bf670318b3db325c71e2ac6a90b7bcfdf9fc739b7cf6aebb31715721623e/intelhex-2.2.1-py2.py3-none-any.whl (50kB)
microbit-vg: Collecting jsonpointer<2.0,>=1.9 (from yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/7b/f4/90449441c1c011a733ee85739a680f2442eda2130e6a88f51804660a0289/jsonpointer-1.14-py2.py3-none-any.whl
microbit-vg: Collecting chardet<3.1.0,>=3.0.2 (from requests[security]<3,>=2.4.3->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/bc/a9/01ffebfb562e4274b6487b4bb1ddec7ca55ec7510b22e4c51f14098443b8/chardet-3.0.4-py2.py3-none-any.whl (133kB)
microbit-vg: Collecting idna<2.7,>=2.5 (from requests[security]<3,>=2.4.3->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/27/cc/6dd9a3869f15c2edfab863b992838277279ce92663d334df9ecf5106f5c6/idna-2.6-py2.py3-none-any.whl (56kB)
microbit-vg: Collecting urllib3<1.23,>=1.21.1 (from requests[security]<3,>=2.4.3->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/63/cb/6965947c13a94236f6d4b8223e21beb4d576dc72e8130bd7880f600839b8/urllib3-1.22-py2.py3-none-any.whl (132kB)
microbit-vg: Collecting pyOpenSSL>=0.14; extra == "security" (from requests[security]<3,>=2.4.3->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/79/db/7c0cfe4aa8341a5fab4638952520d8db6ab85ff84505e12c00ea311c3516/pyOpenSSL-17.5.0-py2.py3-none-any.whl (53kB)
microbit-vg: Collecting MarkupSafe>=0.23 (from Jinja2<3,>=2.7.0->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/4d/de/32d741db316d8fdb7680822dd37001ef7a448255de9699ab4bfcbdf4172b/MarkupSafe-1.0.tar.gz
microbit-vg: Collecting cffi>=1.7; platform_python_implementation != "PyPy" (from cryptography<2.2,>=2.1.4->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/14/dd/3e7a1e1280e7d767bd3fa15791759c91ec19058ebe31217fe66f3e9a8c49/cffi-1.11.5-cp27-cp27mu-manylinux1_x86_64.whl (407kB)
microbit-vg: Collecting enum34; python_version < "3" (from cryptography<2.2,>=2.1.4->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/c5/db/e56e6b4bbac7c4a06de1c50de6fe1ef3810018ae11732a50f15f62c7d050/enum34-1.1.6-py2-none-any.whl
microbit-vg: Collecting asn1crypto>=0.21.0 (from cryptography<2.2,>=2.1.4->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/ea/cd/35485615f45f30a510576f1a56d1e0a7ad7bd8ab5ed7cdc600ef7cd06222/asn1crypto-0.24.0-py2.py3-none-any.whl (101kB)
microbit-vg: Collecting six>=1.4.1 (from cryptography<2.2,>=2.1.4->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/67/4b/141a581104b1f6397bfa78ac9d43d8ad29a7ca43ea90a2d863fe3056e86a/six-1.11.0-py2.py3-none-any.whl
microbit-vg: Collecting ipaddress; python_version < "3" (from cryptography<2.2,>=2.1.4->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/fc/d0/7fc3a811e011d4b388be48a0e381db8d990042df54aa4ef4599a31d39853/ipaddress-1.0.22-py2.py3-none-any.whl
microbit-vg: Collecting functools32; python_version == "2.7" (from jsonschema<3.0,>=2.4.0->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/c5/60/6ac26ad05857c601308d8fb9e87fa36d0ebf889423f47c3502ef034365db/functools32-3.2.3-2.tar.gz
microbit-vg: Collecting pyyaml<4,>=3 (from valinor<1.0,>=0.0.0->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/4a/85/db5a2df477072b2902b0eb892feb37d88ac635d36245a72a6a69b23b383a/PyYAML-3.12.tar.gz (253kB)
microbit-vg: Requirement already satisfied: setuptools in /usr/lib/python2.7/dist-packages (from valinor<1.0,>=0.0.0->yotta) (20.7.0)
microbit-vg: Collecting pyOCD<1.0,>=0.3 (from valinor<1.0,>=0.0.0->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/35/c6/9fee7264155206498796066e924f6edef475b1b683f727e320219b1c85e2/pyOCD-0.8.0.tar.gz (439kB)
microbit-vg: Collecting project_generator<0.9.0,>=0.8.0 (from valinor<1.0,>=0.0.0->yotta)
microbit-vg:   Downloading https://files.pythonhosted.org/packages/b4/bb/91ad20d17485d01829453767323dd2d13b325a0c7154bf5f8f375cbaac1e/project_generator-0.8.16.zip (86kB)
microbit-vg:     Complete output from command python setup.py egg_info:
microbit-vg:     Traceback (most recent call last):
microbit-vg:       File "<string>", line 1, in <module>
microbit-vg:       File "/tmp/pip-install-B7W4Ug/project-generator/setup.py", line 18, in <module>
microbit-vg:         from pip.req import parse_requirements
microbit-vg:     ImportError: No module named req
microbit-vg:
microbit-vg:     ----------------------------------------
microbit-vg: Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-install-B7W4Ug/project-generator/
```

Error is causes by the `project_generator` dependency using pip internals (which is discouraged) that have been moved in the latest v10 released: https://mail.python.org/pipermail/distutils-sig/2017-October/031642.html

Already reported in their issue tracker: https://github.com/project-generator/project_generator/issues/440

Might have to force an older version of pip to be installed until this is resolved.

**Comment by carlosperate** [comment]
Workaround implemented in 5e23b5ed5486a0da44224b49b998df7b0918d950.
I'll leave this open until the all the fixes have been released and I can remove the workaround.

**Comment by carlosperate** [comment]
Workaround removed in d83e77dd5a9d4937016c6d2d8aa4ffff030185ce.
